### DISCUSSION TITLE
CISCO DUO

### DISCUSSION THREAD

**Discussion by SomeDudeTryingtoCode** [akopachov/flipper-zero_authenticator#173] [Q&A]
Will this work with CISCO DUO?

I have attempted multiple times 
https://duo.com/docs/administration-devices#importing-third-party-hardware-tokens

**Comment by akopachov** [comment]
Didn't have a chance to try with Duo. But if Duo supports TOTP algorithm (RFC 6238), then it should work with it.

**Reply by nicholascw** [reply]
Unfortunately, it's not up to the user of Duo to choose the type of 2FA method used. Our system does not enable self-enrollment for OTP devices, nor does it allow WebAuthn.

However, if there's a chance we have standard HOTP, which should be easy given TOTP is very much a special case of HOTP, it should be compatible with Duo Mobile.

**Reply by akopachov** [reply]
Thanks for the details. That's odd that DUO does not allow TOTP however allows HOTP, but if so, then yeah,  it makes sense to add support of HOTP to Authenticator.

Will add to the backlog.

**Reply by akopachov** [reply]
HOTP is implemented in [Flipper Authenticator v5.5.0](https://github.com/akopachov/flipper-zero_authenticator/releases/latest) and [Companion](https://github.com/akopachov/flipper-zero_authenticator-companion/releases/latest)

**Reply by nicholascw** [reply]
![图片](https://github.com/akopachov/flipper-zero_authenticator/assets/7414442/4bf09f41-8260-4a4f-bd2b-018315ba2f14)
Can confirm that they are compatible.

**Reply by akopachov** [reply]
Glad it is working for you!
